BuildAddOn.py

```diff
@@ -173,34 +174,40 @@ def GetProjectGenerationParams (workspaceRootFolder, buildPath, platformName, de
 
     if optionalParams is not None:
         for key in optionalParams:
-            projGenParams.append (f'-D{key}={optionalParams[key]}')
+            if optionalParams[key] is None:
+                projGenParams.append (f'-D{key}')
+            else:
+                projGenParams.append (f'-D{key}={optionalParams[key]}')
 
     projGenParams.append (str (workspaceRootFolder))
 
     return projGenParams
 
 
-def BuildAddOn (configData, platformName, workspaceRootFolder, buildFolder, devKitFolder, version, configuration, languageCode=None):
-    addOnName = configData['addOnName']
-    optionalParams = None
-    if 'addOnSpecificCMakeParameterEnvVars' in configData:
-        optionalParams = {}
-        for key in configData['addOnSpecificCMakeParameterEnvVars']:
-            value = os.environ.get (key)
-            if value is None:
-                raise Exception (f'{key} - Environment variable not found!')
-            optionalParams[key] = value
+def ParseCMakeCache (buildPath):

```

Comment:
This function name explains the implementation details and not the purpose of the function. What about `GetAddOnName`?

Reply:
Function not used anymore